dbt/include/databricks/macros/relations/tblproperties.sql

```diff
@@ -3,8 +3,12 @@
 {%- endmacro -%}
 
 {% macro databricks__tblproperties_clause() -%}
-  {%- set tblproperties = config.get('tblproperties') -%}
-  {%- if tblproperties is not none %}
+  {%- set tblproperties = config.get('tblproperties', {}) -%}
+  {%- set is_iceberg = adapter.check_iceberg(config.get('table_format'), config.get('file_format'), model.config.materialized) -%}
+  {%- if is_iceberg -%}
```

Comment:
This is used by table/view (but not incremental) materialization.  We will fail compilation if the prereqs are violated, but if they are not, and iceberg is set, add the iceberg properties to the dictionary of tblproperties.